### PR TITLE
fix: Add sentiment classification datasets as one multilingual task

### PR DESCRIPTION
**Pull request by imenelydiaker**
Dataset can be fount [here](https://huggingface.co/datasets/mteb/multilingual-sentiment-classification), I added the creation script to mteb/scripts.

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            year = "2023",
+            address = "Dubrovnik, Croatia",
+            publisher = "Association for Computational Linguistics",
+            url = "https://aclanthology.org/2023.sigtyp-1.9",
+            doi = "10.18653/v1/2023.sigtyp-1.9",
+            pages = "89--95",
+        }
+        """,
+        n_samples={"test": 10000},
+        avg_character_length={"test": 45.9},
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py:90]: I'm missing these numbers, I'll update them soon

File: mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "fin": ["fin-Latn"],
+    "hrv": ["hrv-Latn"],
+    "zho": ["zho-Hans"],
+    "cmn": ["cmn-Hans"],
+    "bul": ["bul-Cyrl"],
+    "eus": ["eus-Latn"],
+    "uig": ["uig-Hans"],
+    "bam": ["bam-Latn"],
+    "pol": ["pol-Latn"],
+    # "cym": ["cym-Latn"], # Do not handle this subset since it does not contain a train set required by the evaluation
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py:32]: Could we potentially manually create a train test split via stratified sampling?

File: scripts/data/multilingual_sentiment_classification/create_data.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from datasets import load_dataset
+from huggingface_hub import create_repo
+
+repo_id = "mteb/multilingual-sentiment-classification"
+create_repo(repo_id, repo_type="dataset")
+
+id2label = {0: "negative", 1: "positive"}
```
Comment (isaac-chung) [inline scripts/data/multilingual_sentiment_classification/create_data.py:9]: Unused?

**Review by isaac-chung — COMMENTED** [review]
Looks good at this stage. Don't forget to remove the respective datasets' results as well

File: scripts/data/multilingual_sentiment_classification/create_data.py (hunk 3 above)
Comment (imenelydiaker) [reply scripts/data/multilingual_sentiment_classification/create_data.py:9]: True I'm removing

File: mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py (hunk 2 above)
Comment (imenelydiaker) [reply mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py:32]: I can use `train_test_split` for these `cym` since it contains 25k samples in the test set. 
Although for `hin` the number of samples is under 700 for train and validation combined, I'm not sure it's enough and I'm not comfortable with creating a test split from a train split for a benchmark

Comment (isaac-chung) [reply mteb/tasks/Classification/multilingual/MultilingualSentimentClassification.py:32]: > creating a test split from a train split for a benchmark

I feel the same way. Let's go with what you suggested for `cym`. For `hin` I understand the concern. Is it time consuming to see what performance we get from the just train set as is and eval on the validation set?

**Comment by isaac-chung** [comment]
Looks good. Ready for points?

**Comment by imenelydiaker** [comment]
> Looks good. Ready for points?

I'm adding `cym` train_test_split, didn't have the time since then 😅. On it!

**Comment by imenelydiaker** [comment]
@isaac-chung I added the train split for Welsh (cym), the PR is ready for review 🙂

**Review by isaac-chung — APPROVED** [review]
Looks good. Points then it's ready to merge :)

**Comment by imenelydiaker** [comment]
> Looks good. Points then it's ready to merge :)

Should I add 2 points per language except for thos that were already added in other PRs?

**Comment by isaac-chung** [comment]
It could be 2 pts for the new multilingual dataset, then 4 for each new language not covered in other PRs.

**Comment by imenelydiaker** [comment]
> It could be 2 pts for the new multilingual dataset, then 4 for each new language not covered in other PRs.

ok then it's: 2 + 4 * 4 (Hebrew `heb,` Welsh `cym`, Basque `eus`, Arabic Algerian Dialect `dza`) = 18

**Comment by imenelydiaker** [comment]
@isaac-chung if it's good for you, we should merge then?

**Comment by isaac-chung** [comment]
Oui allons-y!
